jekyll/_cci2/updating-server.adoc

```diff
@@ -5,44 +5,187 @@
+      (context-to-domain context)
+      (context-to-permission context))))
+​
+(defn migrate
+  []
+  (println "===== Syncing orgs =====\n")
+  (sync-all-orgs nil)
+  (println "\n==== Syncing contexts =====\n")
+  (migrate-contexts))
+​
+(migrate)
+```
+
+. If any `ERROR` messages are present in the output please report back to your CSM or reach out to support.
+
+== Update Steps Overview
+
+Following is an overview of the CircleCI Server update steps. Each stage is described in detail below.
+
+* Take a snapshot of your installation so you can rollback later if necessary (optional but recommended)
+* Update Replicated and check you are running Docker v17.12.1, update if necessary
+* Install the latest version of CircleCI Server
+
+=== 1. Snapshot for Rollback
 
 To take a snapshot of your installation:
 
-. Go to the Management Console (e.g. your-circleci-hostname.com:8800) and click Stop Now to stop the CircleCI Services machine from running
-. Ensure no jobs are running on the nomad clients – check by running `nomad status`
+. Go to the Management Console (e.g. `your-circleci-hostname.com:8800`) and click Stop Now to stop the CircleCI service.
++
+.Stop CircleCI
+image::stop_replicated.png[Stop CircleCI]
+. Ensure no jobs are running on the nomad clients – you can check this by running `nomad status`
 . Navigate to the AWS EC2 management console and select your Services machine instance
 . Select Actions > Image > Create Image – Select the No Reboot option if you want to avoid downtime at this point. This image creation step creates an AMI that can be readily launched as a new EC2 instance to restore your installation.
++
+.Snapshot Image Creation
+image::create_snapshot.png[Snapshot Image Creation]
 
 NOTE: It is also possible to automate this process with the AWS API. Subsequent AMIs/snapshots are only as large as the difference (changed blocks) since the last snapshot, such that storage costs are not necessarily larger for more frequent snapshots, see Amazon's EBS snapshot billing document for details.
 Once you have the snapshot you are free to make changes on the Services machine.
 
 If you do need to rollback at any point, see our https://circleci.com/docs/2.0/backup/#restoring-from-backup[guide to restoring from a backup].
 
-== Updating Replicated
+=== 2. Updating Replicated
 
-=== Prerequisites
+==== Prerequisites
 
-* Your installation is Ubuntu 14.04 or 16.04 based.
-* You are running replicated version 2.10.3<= on your services machine. Check by running the following:
+* Your installation is Ubuntu 14.04 or 16.04 based
+* You are running replicated version >=2.10.3 on your services machine. Check by running the following:
 +
 ```shell
 replicated --version
 ```
+// do they need to SSH into the services machine and run this? Actually we need to point them to what to do if their version of Replicated is pre 2.10.3 here
```

Comment:
@GERey there's a few questions for you in comments around here :-)

Reply:
Mostly they need to do this. There is a way for people to possibly upgrade using replicated itself, but I trust the ssh process more than the replicated auto process. A requirement for installing v2 of server is to go to replicated 2.10.3. Nobody on 2.0 should be using a version of replicated before 2.10.3. At least I hope so ;)

Reply:
Done